docs/import/color_function_util.js

```diff
@@ -25,22 +179,136 @@ function withColor(td, layer, property, index) {
       td.append(createColorBox(colorFunction['color']));
       break;
     case ColorStyle.DISCRETE:
-      const colorObject = colorFunction['colors'];
-      const colorSet = new Set();
-      Object.keys(colorObject).forEach((propertyValue) => {
-        const color = colorObject[propertyValue];
-        if (!colorSet.has(color)) {
-          colorSet.add(color);
-          td.append(createColorBox(colorObject[propertyValue]));
-        }
-      });
+      createColorBoxesForDiscrete(colorFunction, td);
       break;
     default:
       setStatus(ILLEGAL_STATE_ERR + 'unrecognized color function: ' + layer);
   }
+  td.on('click', () => onClick(td, colorFunction['current-style']));
   return td;
 }
 
+/**
+ * On click function for color tds - updates the globalTd.
+ * @param {JQuery<HTMLElement>} td
+ * @param {enum} type
+ */
+function onClick(td, type) {
+  if ($(td).hasClass('na')) {
+    return;
+  }
+  const colorFunctionDiv = $('#color-fxn-editor');
+  if (colorFunctionDiv.is(':visible') && td === globalTd) {
+    colorFunctionDiv.hide();
+    return;
+  }
+  colorFunctionDiv.show();
+  if (td === globalTd) {
+    return;
+  }
+  globalTd = td;
+  $('#' + colorStyleTypeStrings.get(type) + '-radio')
+      .prop('checked', true)
+      .trigger('change');
+}
+
+/**
+ * Switches the schema of {@link globalTd} to the given type.
+ * @param {enum} type
+ */
+function switchSchema(type) {
+  const colorFunction = getColorFunction();
+  $('.color-type-div').hide();
+  switch (type) {
+    case ColorStyle.SINGLE:
+      $('#single-color-radio').prop('checked', true);
```

Comment:
nit: Do you actually need to do this? Isn't this triggered by the radio being checked?